port/platform/zephyr/CMakeLists.txt

```diff
@@ -1,18 +1,20 @@
 # This CMake file will select the ubxlib source and include directories
 # to include in the Zephyr application based on Kconfig. Use menuconfig
 # to view the available options.
-#
-# TODO: Instead of compiling the files directly with the app this cmake
-#       file should create an ubxlib library instead.
 if(CONFIG_UBXLIB)
 
+zephyr_library_named(ubxlib)
+
+# we need to be able to include generated header files
+zephyr_include_directories()
+
 get_filename_component(UBXLIB_BASE ../../../ ABSOLUTE)
 set(ENV{UBXLIB_BASE} ${UBXLIB_BASE})
 
 # Add environment variables passed-in via U_FLAGS
 if (DEFINED ENV{U_FLAGS})
     separate_arguments(U_FLAGS NATIVE_COMMAND "$ENV{U_FLAGS}")
-    target_compile_options(app PRIVATE ${U_FLAGS})
```

Comment:
This is an excellent change, thanks for tidying up a "todo", but there's also a pair of `CMakeList.txt` files which form the  `runner` builds that our test system uses, adding-in all of the test files and the test `main()`, see here:

https://github.com/CogniPilot/ubxlib/blob/zephyr-3.4-compile-fix/port/platform/zephyr/runner/CMakeLists.txt

...and here:

https://github.com/CogniPilot/ubxlib/blob/zephyr-3.4-compile-fix/port/platform/zephyr/runner_linux/CMakeLists.txt

Not sure how these should be modified to compile with this change...?

Reply:
FYI, I tried just search/replace `app` with `ubxlib` in those `CMakeLists.txt` files but that resulted in a different error:

```
CMake Error at /home/ubxlib/.ubxlibpkg/nrfconnectsdk-v2.3.0/zephyr/cmake/modules/extensions.cmake:418 (add_library):
  No SOURCES given to target: app
Call Stack (most recent call first):
  /home/ubxlib/.ubxlibpkg/nrfconnectsdk-v2.3.0/zephyr/cmake/modules/kernel.cmake:209 (zephyr_library_named)
  /home/ubxlib/.ubxlibpkg/nrfconnectsdk-v2.3.0/zephyr/cmake/modules/zephyr_default.cmake:117 (include)
  /home/ubxlib/.ubxlibpkg/nrfconnectsdk-v2.3.0/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/ubxlib/.ubxlibpkg/nrfconnectsdk-v2.3.0/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:13 (find_package)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Reply:
Also tried adding a `zephyr_library_named(runner)` line and using that as the target in those `CMakeLists.txt` files but got the same error.